linux/network/interface.sls

```diff
@@ -199,11 +199,30 @@ linux_interface_{{ interface_name }}:
     {%- for network in interface.get('use_ovs_ports', []) %}
     - cmd: ovs_port_up_{{ network }}
     {%- endfor %}
-  {%- endif %}
-  {%- if interface.type == 'bond' %}
+  {%- elif interface.type == 'bond' %}
   - slaves: {{ interface.slaves }}
   - mode: {{ interface.mode }}
-  {%- endif %}
+  - require:
+    {%- for network in interface.get('slaves', '').split() %}
+    - network: linux_interface_{{ network }}
+    {%- endfor %}
+  {%- elif interface.type == 'vlan' %}
+    {%- if interface.get('use_interfaces', []) %}
```

Comment:
cant be [], otherwise it will add empty section `-use:`

Reply:
In python, a empty list will also return False if a list is empty